library/ssl_tls13_server.c

```diff
@@ -325,12 +293,12 @@ static int ssl_tls13_parse_pre_shared_key_ext( mbedtls_ssl_context *ssl,
 
         ret = ssl_tls13_offered_psks_check_identity_match(
                                             ssl, identity, identity_len );
-        if( SSL_TLS1_3_OFFERED_PSK_NOT_MATCH == ret )
+        if( ret != SSL_TLS1_3_OFFERED_PSK_MATCH )
             continue;
 
         ret = ssl_tls13_offered_psks_check_binder_match(
-                                            ssl, binder, binder_len );
-        if( ret != SSL_TLS1_3_OFFERED_PSK_MATCH )
+                                            ssl, binder, binder_len, &alg );
+        if( ret < 0 )
```

Comment:
No, this should be reverted to `ret != SSL_TLS1_3_OFFERED_PSK_MATCH`.
We should validate only one binder.

Spec:
"Prior to accepting PSK key establishment, the server MUST validate
the corresponding binder value (see Section 4.2.11.2 below). If this
value is not present or does not validate, the server MUST abort the
handshake. Servers SHOULD NOT attempt to validate multiple binders;
rather, they SHOULD select a single PSK and validate solely the
binder that corresponds to that PSK. See Section 8.2 and
Appendix E.6 for the security rationale for this requirement."

Reply:
Thanks for identified that. That's my mistake.